apps/api-service/src/transformer.ts

```diff
@@ -1,3 +1,90 @@
+import type { SuperJSONResult } from "superjson";
 import { SuperJSON } from "superjson";
 
-export const transformer = SuperJSON;
+/**
+ * Creates a transformer that wraps SuperJSON and warns about unsupported data types.
+ * Specifically, it detects class instances that haven't been registered with SuperJSON,
+ * as these cannot be properly serialized and may cause issues.
+ */
+function createWarningTransformer() {
+  const warnedTypes = new Set<string>();
+
+  /**
+   * Recursively checks for class instances in the data structure.
+   * Class instances are objects whose constructor is not one of the built-in types.
+   */
+  function detectUnsupportedTypes(value: unknown, path = "root"): void {
```

Comment:
don't define this function inside createWarningTransformer, move it out to top level